---
title: C6320
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: fef29c7f6e341cf4cd31991b24bfc209bd902b15
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 04/23/2019
ms.locfileid: "62781473"
---
# <a name="c6320"></a>C6320
경고 C6320: 예외 필터 식이 상수 EXCEPTION_EXECUTE_HANDLER입니다. 이 처리 의도 하지 않은 예외를 마스킹할 수 있습니다.

 이 경고는 영향을 주는 __except 블록에서 EXCEPTION_EXECUTE_HANDLER 상수를 사용 하 여 나타냅니다. 이 경우 문에서 \__except 블록 보내지 않으려는 특정 함수를 처리 하는 예외를 포함 하 여 예외를 처리 하려면 항상 실행 됩니다. 처리 하기 전에 예외를 확인 하는 것이 좋습니다.

## <a name="example"></a>예제
 __Except은 차단 하기 때문에이 경고는 다음 코드에서는 모든 종류의 예외를 처리 하려고 합니다.

```
#include <stdio.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;  // might cause access violation exception
      // code ...
   }
   __except(EXCEPTION_EXECUTE_HANDLER) // warning
   {
      puts("in except");
      // code ...
   }
}
```

 이 경고를 해결 하려면 사용 하 여 `GetExceptionCode` 다음 코드에 나와 있는 것 처럼 처리 하기 전에 특정 예외를 확인 하려면:

```
#include <stdio.h>
#include <windows.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;    // might cause access violation exception
      // code ...
   }
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
   {
      puts("in except");
      // code ...
   }
}
```

## <a name="see-also"></a>참고 항목
 [try-except 문](/cpp/cpp/try-except-statement)